5.7.36 codership mysql wsrep 5.7.35 25.27

nitendra-bhosle opened the pull request.

ayurchen committed:
Avoid code duplication and make sure GTID is streamed in identical way
be it SST or bypass.

Refs codership/mysql-wsrep-bugs#943

ayurchen committed:
Enable configurable timeouts for donor and joiner: options donor-timeout
and joiner-timeout in the [sst] config section. Default to 10 and 60 seconds
respectively.

Add listenining daemon cleanup for xtrabackup-v2 SST for the cases when
timeout is 0 (no timeout)

Refs codership/mysql-wsrep-bugs##936

temeo committed:
Merge tag 'mysql-5.7.35' into 5.7-v25-5.7.35

Commit:
Lock `LOCK_wsrep_thd` should only protect access to `wsrep_conflict_state`

LOCK_wsrep_thd should be used to protect acces to `wsrep_conflict_state` variable.
Release lock once variable is used.

Commit:
add new wsrep_mode IGNORE_CASCADING_FK_DELETE_MISSING_ROW_ERROR

Work around mysql bug #80821. On cascading deletes, when both parent and child
are involved in the query, a delete rows event for the child may be added to the
binlog after the parent (cascading) delete. This will cause the delete to fail
on slaves. This commit causes such errors to be ignored on slaves when wsrep_mode
includes the new option IGNORE_CASCADING_FK_DELETE_MISSING_ROW_ERROR

The test galera.galera_fk_multitable was affected by the bug and has been
re-enabled with the warning suppressed.

Commit:
codership/mysql-wsrep#364 Multimaster hang with innodb-thread-concurrency=8

Access to `thd->wsrep_conflict_state` should be protected with mutex `LOCK_wsrep_thd`.
Applier and TOI threads should no be affected with usage of `innodb_thread_concurrency`
variable so returning before any checks.

Commit:
Allow FLUSH TABLE when node is in detached state

abychko committed:
GitHub #1074 Fix cpack source package generation

added wsrep version to source tarball, packed submodules

abychko committed:
Merge pull request #1080 from codership/5.7-v25-1074

GitHub #1074 Fix cpack source package generation

temeo committed:
Merged man pages from upstream 5.7.35

temeo committed:
Bumped wsrep patch version to 25.27 for 5.7.35

abychko committed:
removed --exclude-vcs-ignores option to tar because it produced
incomplete source tarball

nitendra-bhosle commented:
Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/262/

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1578) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1578) before we can accept your contribution.<br/>**0** out of **4** committers have signed the CLA.<br/><br/>:x: abychko<br/>:x: ayurchen<br/>:x: mkaruza<br/>:x: temeo<br/><hr/>**mkaruza** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster?pullRequest=1578) it.</sub>

nitendra-bhosle commented:
Latest Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/264/

venkatesh-prasad-v commented on scripts/wsrep_sst_rsync.sh:
```diff
@@ -178,6 +178,21 @@ FILTER=(-f '- /lost+found'
 if [ "$WSREP_SST_OPT_ROLE" = "donor" ]
 then
 
+||||||| merged common ancestors
```
Version Control Marker will cause rsync tests to fail. Please remove this.


>2022-01-19T07:14:07.489217Z 0 [Note] WSREP: Initiating SST/IST transfer on JOINER side (wsrep_sst_rsync --role 'joiner' --address '127.0.0.1:13009' --datadir '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/mysqld.2/data/' --defaults-file '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/my.cnf' --defaults-group-suffix '.2' --parent '39359' --mysqld-version '5.7.36-39'   '' ) 
/home/venki/work/pxc/review/nitendra/bld/scripts/wsrep_sst_rsync: line 181: syntax error near unexpected token `||'
>2022-01-19T07:14:07.661565Z 0 [ERROR] WSREP: Failed to read 'ready <addr>' from: wsrep_sst_rsync --role 'joiner' --address '127.0.0.1:13009' --datadir '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/mysqld.2/data/' --defaults-file '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/my.cnf' --defaults-group-suffix '.2' --parent '39359' --mysqld-version '5.7.36-39'   '' 
	Read: '(null)'

venkatesh-prasad-v commented on scripts/wsrep_sst_rsync.sh:
```diff
@@ -178,6 +178,21 @@ FILTER=(-f '- /lost+found'
 then
 
+||||||| merged common ancestors
+cat << EOF > "$STUNNEL_CONF"
+key = $SSTKEY
+cert = $SSTCERT
+${CAFILE_OPT}
+foreground = yes
+pid = $STUNNEL_PID
+debug = warning
+client = yes
+connect = ${WSREP_SST_OPT_HOST_UNESCAPED}:${WSREP_SST_OPT_PORT}
+TIMEOUTclose = 0
+${VERIFY_OPT}
+EOF
+#connect = ${WSREP_SST_OPT_ADDR%/*}
```
We reverted Codership's stunnel encryption in rsync by commit 998da9b5d. So this block can completely be removed.

venkatesh-prasad-v commented on sql/log_event.cc:
```diff
@@ -10500,6 +10500,30 @@ int Rows_log_event::handle_idempotent_and_ignored_errors(Relay_log_info const *r
                          ignored_error_code(actual_error) : 0);
+#if defined(WITH_WSREP)
+    /*
+     * Work around mysql Bug#80821. With cascading deletes, when both parent and
+     * child tables are involved in a multi table query, a delete rows event for
+     * the child may be added to the binlog after the parent (cascading) delete.
+     * This will cause the delete to fail on slaves. A new wsrep_mode value has
+     * been added for ignoring such delete errors.
+     */
```
IIUC, [Bug#80821](https://bugs.mysql.com/bug.php?id=80821) is already fixed in PS and PXC. So I'm not sure if this is really necessary.

@dutow Can you please share your opinions on this?

Commit:
PXC-3788: Merge codership mysql-wsrep-5.7.35-25.27 Part 1 (with conflicts)

Files having merge conflicts:
cmake/mysql_version.cmake
cmake/wsrep.cmake
scripts/wsrep_sst_rsync.sh
scripts/wsrep_sst_xtrabackup-v2.sh
sql/CMakeLists.txt
sql/sql_parse.cc
storage/innobase/handler/ha_innodb.cc

Merge tag 'wsrep_5.7.35-25.27' into 5.7.36-codership-mysql-wsrep-5.7.35-25.27

Commit:
PXC-3788: Merge codership/wsrep_5.7.35-25.27 -> percona/pxc-5.7.36 (conflicts resolved)

https://jira.percona.com/browse/PXC-3788

Merge conflicts resolved

***
Updated WSREP_PATCH_VERSION to 55 in the wsrep.cmake file.

***
Updated WSREP-REVISION to reflect the upstream refresh revision
commit id: 3610f79

nitendra-bhosle commented:
New Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/267/

nitendra-bhosle commented:
Jenkins: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/268/

venkatesh-prasad-v commented on sql/log_event.cc:
```diff
@@ -10500,6 +10500,30 @@ int Rows_log_event::handle_idempotent_and_ignored_errors(Relay_log_info const *r
                            (rbr_exec_mode == RBR_EXEC_MODE_IDEMPOTENT));
     bool ignored_error= (idempotent_error == 0 ?
                          ignored_error_code(actual_error) : 0);
+#if defined(WITH_WSREP)
```
Please change it to `#ifdef WITH_WSREP` in order to make it work with our `check_src` tool.

venkatesh-prasad-v commented on sql/sql_parse.cc:
```diff
@@ -6379,13 +6380,7 @@ case SQLCOM_PREPARE:
   mysql_mutex_lock(&thd->LOCK_wsrep_thd);
   if (thd->wsrep_conflict_state != REPLAYED)
   {
-  /* Hold this lock only for duration of wsrep_conflict_state state check.
-  Release it before entering unit-cleanup. Unit-cleanup will cause
-  external_unlock at InnoDB level that will need transaction lock.
-  This could conflict if this thread has conflicting lock with
-  other high-priority thread that needs transaction lock and
-  LOCK_wsrep_thd too. */
```
Please revert this change (bring back the comment).

venkatesh-prasad-v reviewed (approved):
lgtm

Commit:
PXC-3788: Merge codership/wsrep_5.7.35-25.27 -> percona/pxc-5.7.36

https://jira.percona.com/browse/PXC-3788

Updated submodule (percona-xtradb-cluster-galera) pointer to 3.x-5.7
